src/thermo/PlasmaPhase.cpp

```diff
@@ -272,6 +303,177 @@ void PlasmaPhase::initThermo()
     }
 }
 
+void PlasmaPhase::addSolution(std::weak_ptr<Solution> soln) {
+    ThermoPhase::addSolution(soln);
+    // register callback function to be executed
+    // when the thermo or kinetics object changed
+    if (shared_ptr<Solution> soln = m_soln.lock()) {
+        soln->registerChangedCallback(this, [&]() {
+            setCollisions();
+        });
+    }
+}
+
+void PlasmaPhase::setCollisions()
+{
+    m_collisions.clear();
+    m_targetSpeciesIndices.clear();
+
+    if (shared_ptr<Solution> soln = m_soln.lock()) {
+        shared_ptr<Kinetics> kin = soln->kinetics();
+        if (!kin) return;
```

Comment:
Please don't use the one-line version of the `if` statement -- always use braces and put the body on a separate line.